### PR TITLE
[CLD-4490] Add support to enable aws service devops guru

### PR DESCRIPTION
**Pull request by andrleite**
#### Summary

- Add support to enable AWS Devops Guru for database instances

#### Ticket Link
 Fixes https://mattermost.atlassian.net/browse/CLD-4490

#### Release Note

```release-note
- Add support to enable aws devops guru service
```

### CODE REVIEW

File: cmd/dbfactory/cluster.go [hunk 1]
```diff
@@ -65,6 +66,7 @@ var clusterProvisionCmd = &cobra.Command{
 		replicas, _ := command.Flags().GetString("replicas")
 		dbProxy, _ := command.Flags().GetBool("dbproxy")
 		creationSnapshotARN, _ := command.Flags().GetString("creation-snapshot-arn")
+		enableDevopsGuru, _ := command.Flags().GetBool("devopsguru")
```
Comment (d-wierdsma) [inline cmd/dbfactory/cluster.go:69]: Let's do `devops-guru` instead of `devopsguru`

File: cmd/dbfactory/cluster.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ func init() {
 	clusterProvisionCmd.Flags().String("replicas", "3", "The total number of write/read replicas.")
 	clusterProvisionCmd.Flags().Bool("dbproxy", true, "If enabled the multitenant DB cluster will be used with a DB proxy.")
 	clusterProvisionCmd.Flags().String("creation-snapshot-arn", "", "The ARN of the snapshot to use for the DB cluster (default \"\")")
+	clusterProvisionCmd.Flags().Bool("devopsguru", false, "Enable the AWS service Devops Guru to all database instances within a cluster.")
```
Comment (d-wierdsma) [inline cmd/dbfactory/cluster.go:33]: Let's do `devops-guru` instead of `devopsguru`

File: terraform/aws/modules/rds-aurora-postgresql/main.tf [hunk 3]
```diff
@@ -99,9 +99,8 @@ resource "aws_rds_cluster_instance" "provisioning_rds_db_instance" {
       "DatabaseType"                        = var.multitenant_tag,
       "MattermostCloudInstallationDatabase" = "PostgreSQL/Aurora"
     },
-    var.tags
-  )
-
+    var.tags,
+    [var.enable_devops_guru ? {"devops-guru-default" = aws_rds_cluster.provisioning_rds_cluster.cluster_identifier} : null]...)
```
Comment (d-wierdsma) [inline terraform/aws/modules/rds-aurora-postgresql/main.tf:103]: Nice one with the `...)` :)

File: api/api.go [hunk 4]
```diff
@@ -27,19 +27,20 @@ func initCluster(apiRouter *mux.Router, context *Context) {
...
+//		  "environment": "dev",
+//		  "stateStore": "terraform-database-factory-state-bucket-dev",
+//		  "apply": true,
+//		  "instanceType": "db.r5.large",
+//		  "clusterID": "12345678",
+//		  "backupRetentionPeriod": "15",
+//		  "dbEngine: postgres",
+//		  "replicas": "3",
+//		  "dbProxy": true,
+//		  "creationSnapshotARN": "",
```
Comment (stafot) [inline api/api.go:42]: Add the new variable on the sample body too

File: cmd/dbfactory/cluster.go (hunk 1 above)
Comment (stylianosrigas) [reply cmd/dbfactory/cluster.go:69]: +1 on this ;)

**Review by stylianosrigas — APPROVED** [review]

**Review by stylianosrigas — APPROVED** [review]
LGTM

**Pull request merged**